common/src/main/java/org/apache/rocketmq/common/topic/TopicValidator.java

```diff
@@ -62,14 +58,39 @@
         SYSTEM_TOPIC_SET.add(RMQ_SYS_OFFSET_MOVED_EVENT);
 
         NOT_ALLOWED_SEND_TOPIC_SET.add(RMQ_SYS_SCHEDULE_TOPIC);
+
+        // regex: ^[%|a-zA-Z0-9_-]+$
+        // %
+        VALID_CHAR_BIT_MAP['%'] = true;
+        // -
+        VALID_CHAR_BIT_MAP['-'] = true;
+        // _
+        VALID_CHAR_BIT_MAP['_'] = true;
```

Comment:
is the '|' missing?  the original regex contains '|'.

Reply:
oh it's my fault, I used to think it was a separator. I have fixed it.